src/main/java/org/openstreetmap/atlas/checks/validation/areas/TallBuildingCheck.java

```diff
@@ -0,0 +1,723 @@
+            ":", ";", "<", ",", ">", "?", "/");
+    private static final int INSTRUCTION_THREE = 3;
+    private static final int INSTRUCTION_FOUR = 4;
+    private static final double ONE_QUARTER = 0.25;
+    private static final double THREE_QUARTERS = 0.75;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public TallBuildingCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.bufferDistanceMeters = this.configurationValue(configuration, "buffer.distance.meters",
+                BUFFER_DISTANCE_DEFAULT);
+        this.minDatasetSizeForStatsComputation = this.configurationValue(configuration,
+                "min.dataset.size.for.stats.computation", MIN_DATASET_SIZE_DEFAULT);
+        this.maxLevelTagValue = this.configurationValue(configuration, "max.level.tag.value",
+                MAX_LEVEL_TAG_VALUE_DEFAULT);
+        this.outlierMultiplier = this.configurationValue(configuration, "outlier.multiplier",
+                OUTLIER_MULTIPLIER_DEFAULT);
+        this.invalidHeightCharacters = new HashSet<>(this.configurationValue(configuration,
+                "invalid.height.characters", INVALID_CHARACTER_DEFAULT));
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        final Map<String, String> tags = object.getTags();
+        return !this.isFlagged(object.getOsmIdentifier())
+                && ((object instanceof Area)
+                        || (object instanceof Relation && ((Relation) object).isMultiPolygon()))
+                && (this.isBuildingOrPart(object) || this.isBuildingRelationMember(object))
+                && (this.hasHeightTag(tags) || this.hasBuildingLevelTag(tags));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        markAsFlagged(object.getOsmIdentifier());
+        final Map<String, String> tags = object.getOsmTags();
+
+        // level tag logic
+        if (this.hasBuildingLevelTag(tags))
```

Comment:
any undesirable outcome if the object has a buildingLevel tag and a height tag?

Reply:
the first scenario that the feature fits under, if any, short circuits the flag method even if the feature might fit under other scenarios. Should I try to catch all scenarios with the flag?

Reply:
addressed offline